examples/rdl.py

```diff
@@ -0,0 +1,710 @@
+
+        # Get the target column name from the task
+        loss = loss_fn(pred.float(), batch[task.entity_table].y.float())
+        loss.backward()
+        optimizer.step()
+
+        loss_accum += loss.detach().item() * pred.size(0)
+        count_accum += pred.size(0)
+
+    return loss_accum / count_accum
+
+
+@torch.no_grad()
+def test(loader: NeighborLoader, model: Model, task: EntityTask) -> np.ndarray:
+    model.eval()
+
+    pred_list = []
+    for batch in loader:
+        batch = batch.to(device)
+        pred = model(
+            batch,
+            task.entity_table,
+        )
+        pred = pred.view(-1) if pred.size(1) == 1 else pred
+        pred_list.append(pred.detach().cpu())
+    return torch.cat(pred_list, dim=0).numpy()
+
+
+if __name__ == "__main__":
+    """ Prepare the data """
+
+    # Load the dataset
+    print("Loading dataset...")
+    dataset = get_dataset(name=args.dataset, download=True)
+
+    # Load the task
+    print("Loading task...")
+    task = get_task(dataset_name=args.dataset, task_name=args.task,
+                    download=True)
+
+    # Print task information for debugging
+    print(f"Task type: {task.task_type}")
+    print(f"Target column: {task.target_col}")
+    print(f"Entity table: {task.entity_table}")
+
+    # get the column to stype dictionary
+    print("Getting column to stype dictionary...")
+    db = dataset.get_db()
+    col_to_stype_dict = get_stype_proposal(db)
+    print("Column to stype dictionary: ", col_to_stype_dict)
+
+    # define the text embedder
+    print("Defining text embedder...")
+    text_embedder_cfg = TextEmbedderConfig(
+        text_embedder=GloveTextEmbedding(device=device), batch_size=256)
+
+    # Transform the dataset into a HeteroData object with torch_frame features
+    print("Transforming dataset into HeteroData object...")
+    data, col_stats_dict = make_pkey_fkey_graph(
+        db,
+        col_to_stype_dict=col_to_stype_dict,  # speficied column types
+        text_embedder_cfg=text_embedder_cfg,  # our chosen text encoder
+        cache_dir=os.path.join(
+            root_dir, f"{args.dataset}_{args.task}_materialized_cache"
+        ),  # store materialized graph for convenience
+    )
+    """ Prepare data loaders """
+    print("Preparing data loaders...")
+    loader_dict = {}
+
+    # load the training, validation, and test tables
+    train_table = task.get_table("train")
+    val_table = task.get_table("val")
+    test_table = task.get_table("test")
+
+    # Print table information for debugging
+    print(f"Train table columns: {train_table.df.columns}")
+    print(f"Val table columns: {val_table.df.columns}")
+    print(f"Test table columns: {test_table.df.columns}")
+
+    for split, table in [
+        ("train", train_table),
+        ("val", val_table),
+        ("test", test_table),
+    ]:
+        table_input = get_node_train_table_input(
+            table=table,
+            task=task,
+        )
```

Comment:
Can we expose this logic here in the example? That's pretty crucial for the understanding.